### PR TITLE
Fix hmac secret

### PR DESCRIPTION
**Pull request by zhaohuabing**
This PR does the following:
* Generates an HMAC secret and stores it in a K8s secret in the CertGen job.
* Watches this k8s secret in the Kuberetes provider.
* Uses the HMAC generated in the above step for envoy ouath2 filter configuration.

### CODE REVIEW

File: internal/crypto/certgen.go [hunk 1]
```diff
@@ -153,6 +154,11 @@ func GenerateCerts(cfg *config.Server) (*Certificates, error) {
 			return nil, err
 		}
 
+		oidcHMACSecret, err := generateHMACSecret()
```
Comment (arkodg) [inline internal/crypto/certgen.go:157]: what happens during upgrade if this secret already exists ? do we skip or overwrite ?

Comment (arkodg) [reply internal/crypto/certgen.go:157]: relates to https://github.com/envoyproxy/gateway/blob/ce1eb544d5e1b473ccd2c852be595d80fcf75ba8/internal/provider/kubernetes/secrets.go#L84

**Comment by arkodg** [comment]
hey @zhaohuabing has this been tested ? are you sure we are reconciling this secret ? I think we may need some more code in https://github.com/envoyproxy/gateway/blob/ce1eb544d5e1b473ccd2c852be595d80fcf75ba8/internal/provider/kubernetes/controller.go#L138 to make sure this secret is part of the resources

File: internal/crypto/certgen.go (hunk 1 above)
Comment (zhaohuabing) [reply internal/crypto/certgen.go:157]: The creation of HMAC secret will be skipped in `CreateOrUpdateSecrets` if it's already there, like the other certs. No changes are needed.

**Comment by zhaohuabing** [comment]
> hey @zhaohuabing has this been tested ? are you sure we are reconciling this secret ?
I haven't tested it manually yet. It should be covered by the OIDC e2e test: https://github.com/envoyproxy/gateway/pull/2730/

> I think we may need some more code in
> 
> https://github.com/envoyproxy/gateway/blob/ce1eb544d5e1b473ccd2c852be595d80fcf75ba8/internal/provider/kubernetes/controller.go#L138
> 
> to make sure this secret is part of the resources

Yes, I missed that.  It's fixed.

**Review by Xunzhuo — APPROVED** [review]

**Review by arkodg — APPROVED** [review]
🚀

**Pull request merged**